src/types/zkgyaml_eth.ts

```diff
@@ -105,7 +109,7 @@ export class EthereumDataSource extends DataSource {
 
   // signaficant to decide which lib dsp main it should use.
   getSignificantKeys() {
```

Comment:
Should give here a correct type.
e.g.
```ts
getSignificantKeys(): [DataSourceKind, 'unsafe'] | [DataSourceKind] {
  return this.unsafe ? [this.kind, 'unsafe'] : [this.kind]
}
```
or
```ts
getSignificantKeys(): [DataSourceKind, 'unsafe' | undefined] {
  return this.unsafe ? [this.kind, 'unsafe'] : [this.kind, undefined]
}
```

Reply:
`string[]`: unequal length string array, slightly ambiguous

Reply:
added in 867a051